docs/src/main/asciidoc/pubsub.adoc

```diff
@@ -25,94 +25,130 @@ dependencies {
-Other implementations of the `PubSubMessageConverter` can also be configured in the same manner.
+[cols="20,80"]
+|===
+h|subscribe(String subscription, Consumer<BasicAcknowledgeablePubsubMessage> messageConsumer) | asynchronously pulls messages and passes them to `messageConsumer`
+h|subscribeAndConvert(String subscription,
+ 			Consumer<ConvertedBasicAcknowledgeablePubsubMessage<T>> messageConsumer,
+ 			Class<T> payloadType) | same as `pull`, but converts message payload to `payloadType` using the converter configured in the template
+|===
 
-Please refer to our link:../spring-cloud-gcp-samples/spring-cloud-gcp-integration-pubsub-json-sample[Pub/Sub JSON Payload Sample App] as a reference for using this functionality.
 
-==== Subscribing to a subscription
+==== Pulling messages from a subscription
 
-Google Cloud Pub/Sub allows many subscriptions to be associated to the same topic.
-`PubSubTemplate` allows you to listen to subscriptions via the `subscribe()` method.
-It relies on a `SubscriberFactory` object, whose only task is to generate Google Cloud Pub/Sub
-`Subscriber` objects.
-When listening to a subscription, messages will be pulled from Google Cloud Pub/Sub
-asynchronously, at a certain interval.
+Google Cloud Pub/Sub supports synchronous pulling of messages from a subscription.
+This is different from subscribing to a subscription, in the sense that subscribing is an asynchronous task.
 
-The Spring Boot starter for Google Cloud Pub/Sub auto-configures a `SubscriberFactory`.
+===== Example
+Pull up to 10 messages:
+[source,java,indent=0]
+----
+include::../../../src/test/java/org/springframework/cloud/gcp/autoconfigure/pubsub/it/PubSubTemplateDocumentationTests.java[tag=pull]
+----
 
-If Pub/Sub message payload conversion is desired, you can use the `subscribeAndConvert()` method, which will use the converter configured in the template.
+===== Pull methods
 
-==== Pulling messages from a subscription
+`PubsubTemplate` provides the following pull methods:
+[cols="20,80"]
+|===
+h|pull(String subscription, Integer maxMessages,
+       			Boolean returnImmediately) | Pulls a number of messages from a subscription, allowing for the retry settings to be configured.
+       			Any messages received by `pull()` are not automatically acknowledged. See <<Acknowledging messages>>.
 
-Google Cloud Pub/Sub supports synchronous pulling of messages from a subscription.
-This is different from subscribing to a subscription, in the sense that subscribing is an asynchronous task which polls the subscription on a set interval.
+       			If `returnImmediately` is `true`, the system will respond immediately even if it there are no messages available to return in the `Pull` response. Otherwise, the system may wait (for a bounded amount of time) until at least one message is available, rather than returning no messages.
+h|pullAndAck | Works the same as the `pull` method and, additionally, acknowledges all received messages.
```

Comment:
I think for consistency, the method names should either all have their methods parameters or all not have method parameters. Right now because `pull` has the method parameters, it feels like the other methods have the same method signature as `pull`; however this is not the case.